pylint/config/option_manager_mixin.py

```diff
@@ -271,6 +271,9 @@ def read_config_file(self, config_file=None, verbose=None):
 
         use_config_file = config_file and os.path.exists(config_file)
         if use_config_file:
+            mod_name = config_file
+            self.current_name = mod_name
+            self.set_current_module(mod_name)
```

Comment:
This is required to not crash on #4746 regression test.

Reply:
Is this what you're suggesting to do ?
````diff
diff --git a/pylint/config/option_manager_mixin.py b/pylint/config/option_manager_mixin.py
index 924998e7..3595d466 100644
--- a/pylint/config/option_manager_mixin.py
+++ b/pylint/config/option_manager_mixin.py
@@ -271,9 +271,8 @@ class OptionsManagerMixIn:
 
         use_config_file = config_file and os.path.exists(config_file)
         if use_config_file:
-            mod_name = config_file
-            self.current_name = mod_name
-            self.set_current_module(mod_name)
+            self.current_name = config_file
+            self.set_current_module(self.current_name)
             parser = self.cfgfile_parser
             if config_file.endswith(".toml"):
                 self._parse_toml(config_file, parser)
```

Reply:
See my latest comment!